docs/getting-started-guides/mesos.md

```diff
@@ -0,0 +1,303 @@
+
+Running Kubernetes on Mesos allows you to easily move Kubernetes workloads from one cloud provider to another to your own physical datacenter.
+
+This tutorial will walk you through setting up Kubernetes on a Mesos cluster on [Google Cloud Plaform][4].
+It provides a step by step walk through of adding Kubernetes to a Mesos cluster and running the classic GuestBook demo application.
+The walkthrough presented here is based on the v0.4.x series of the Kubernetes-Mesos project, which itself is based on Kubernetes v0.11.0.
+
+### Prerequisites
+
+* Understanding of [Apache Mesos][11]
+* Mesos cluster on [Google Compute Engine][5]
+* Identify the Mesos master node external IP from Mesosphere [cluster launch pad][12]
+* A [VPN connection to the cluster][6], or setup firewall rules for direct access (used in this doc).
+
+### Setup Access to Mesos Admin GUI
+This doc will refer to the running Mesos cluster repeatedly.  It's a good idea to make the Mesos admin GUI accessible via the browser.  First, take note of the network name attached to your Mesos cluster (either via the Google Cloud GUI) or with `gcloud` tool as follows:
+
+```bash
+gcloud compute firewall-rules list
+```
+Next, add a firewall-rule to open traffic on port 5050.  This will allow you to access Mesos master GUI via the browser.
+
+```bash
+gcloud compute firewall-rules create mesos-master-gui --network ${network_name}  --source-ranges 0.0.0.0/0 --allow tcp:5050
+```
+Once the rule is applied, point your browser to your Mesos master's IP address (see pre-requisites).  You should see the Mesos GUI displayed.
+
+### Deploy Kubernetes-Mesos
+
+Log into the master node over SSH, replacing the placeholder below with the correct IP address.
+
+```bash
+ssh jclouds@${ip_address_of_master_node}
+```
+
+Build Kubernetes-Mesos.
+
+```bash
+$ git clone https://github.com/mesosphere/kubernetes-mesos k8sm
+$ mkdir -p bin && sudo docker run --rm -v $(pwd)/bin:/target \
+  -v $(pwd)/k8sm:/snapshot -e GIT_BRANCH=release-0.4 \
+  mesosphere/kubernetes-mesos:build
+```
+
+Set some environment variables.
+The internal IP address of the master is visible via the cluster details page on the Mesosphere launchpad, or may be obtained via `hostname -i`.
+
+```bash
+$ export servicehost=$(hostname -i)
+$ export mesos_master=${servicehost}:5050
+$ export KUBERNETES_MASTER=http://${servicehost}:8888
+```
+### Deploy etcd
+Start etcd and verify that it is running:
+
+```bash
+$ sudo docker run -d --hostname $(hostname -f) --name etcd -p 4001:4001 -p 7001:7001 coreos/etcd
+```
+
+```bash
+$ sudo docker ps
+CONTAINER ID   IMAGE                COMMAND   CREATED   STATUS   PORTS                NAMES
+fd7bac9e2301   coreos/etcd:latest   "/etcd"   5s ago    Up 3s    2379/tcp, 2380/...   etcd
+```
+It's also a good idea to ensure your etcd instance is reachable by testing it
+```bash
+curl -L http://$(hostname -i):4001/v2/keys/
```

Comment:
any reason why `$(hostname -i)` is used here instead of `$servicehost`? I think it should be consistent.